tests/wandb_run_test.py

```diff
@@ -300,3 +300,42 @@ def test_unlogged_artifact_in_config(live_mock_server, test_settings):
             str(e_info.value)
             == "Cannot json encode artifact before it has been logged or in offline mode."
         )
+
+
+def test_anonymous_mode(live_mock_server, test_settings, capsys):
+    api_key = os.environ.pop("WANDB_API_KEY")
+    uname = os.environ.pop("WANDB_USERNAME")
+    test_settings.update({"anonymous": "must", "api_key": None})
```

Comment:
Does this persist after the test? It would be safer to copy the settings instead of updating it in place as otherwise other tests could observe the effects of this update.

Reply:
These changes don't persist after the test, and its a common pattern in our cli tests.

Reply:
Gotcha, thanks! Is that handled by the pytest fixtures? Trying to understand how the env gets reset each test.

Reply:
Oh, yeah for `test_settings` pytest fixtures handles the resetting of the value. But the environment is not always reset between tests. We've switched over to using a patched environment context as you suggested. Great idea!